src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -278,4 +314,14 @@ private function addServiceToAmbiguousType($id, $type)
         }
         $this->ambiguousServiceTypes[$type][] = $id;
     }
+
+    static private function getResourceMetadataForMethod(\ReflectionMethod $method)
+    {
+        $methodArgumentsMetadata = array();
+        foreach ($method->getParameters() as $parameter) {
+            $methodArgumentsMetadata[] = (string) $parameter;
```

Comment:
@fabpot a change in the default value also affects autowiring when the default value must be used